sql/core_clients_daily_v1.sql

```diff
@@ -0,0 +1,95 @@
+CREATE TEMP FUNCTION udf_mode_last(x ANY TYPE) AS ((
+  SELECT
+    val
+  FROM (
+    SELECT
+      val,
+      COUNT(val) AS n,
+      MAX(offset) AS max_offset
+    FROM
+      UNNEST(x) AS val
+    WITH OFFSET AS offset
+    GROUP BY
+      val
+    ORDER BY
+      n DESC,
+      max_offset DESC
+  )
+  LIMIT 1
+));
+
+WITH
+  numbered_duplicates AS (
+  SELECT
+    * REPLACE(LOWER(client_id) AS client_id),
+    ROW_NUMBER() OVER (PARTITION BY client_id, submission_date_s3, metadata.document_id ORDER BY metadata.timestamp DESC) AS n
```

Comment:
I thought I saw a PR float by where an "internal-use" type field was aliased to begin with an underscore - I thought that was a nice idea, so that might be a good thing to use here for a temporary field we remove in the next CTE step, just in case someone adds an `n` field to the core ping.

Reply:
@relud has advocated for this, too. You two are probably right. I've just seen a lot of SQL queries using ROW_NUMBER and WHERE n = 1, so it feels right to me. I have no proof this is a wider standard in the community, though, and you make a good point about potentially running into difficulty with a field named `n` in the source table.

Breaking out this discussion to https://github.com/mozilla/bigquery-etl/issues/10